### PR TITLE
[DOC] Update Set up test app to use agent flow

### LINKED ISSUE #3226: [DOC] Update the Set up a test app doc to use Agent Flow
The [Set up an example test app](https://grafana.com/docs/tempo/latest/setup/set-up-test-app/) uses Agent static mode and needs to be updated.

### PR DESCRIPTION
**Pull request by knylander-grafana**
**What this PR does**:

Updates the Set up the test app doc to use Agent Flow and Intro to MLTP as an option. 

**Which issue(s) this PR fixes**:
Fixes #3226 

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/tempo/setup/set-up-test-app.md [hunk 1]
```diff
@@ -15,110 +15,102 @@ For example, if you [set up Tempo using the Kubernetes with Tanka procedure]({{<
...
-                          - __meta_kubernetes_pod_container_name
-                        target_label: container
-                    tls_config:
-                        ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
-                        insecure_skip_verify: false
-    ```
+## Configure Grafana Agent Flow to remote-write to Tempo
 
-    If you have followed the [Tanka Tempo installation example]({{< relref "../setup/tanka" >}}), then the `endpoint` value would be:
+We'll use a [Grafana Agent Helm chart](https://grafana.com/docs/agent/latest/flow/setup/install/kubernetes/) deployment to send traces to Tempo.
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/set-up-test-app.md:27]:
```suggestion
We'll use a [Grafana Agent Helm chart](/docs/agent/latest/flow/setup/install/kubernetes/) deployment to send traces to Tempo.
```

**Review by joe-elliott — COMMENTED** [review]
Couple o thoughts. I didn't test any of the actual config or steps. Trust these are correct.

File: docs/sources/tempo/setup/set-up-test-app.md [hunk 2]
```diff
@@ -15,110 +15,102 @@ For example, if you [set up Tempo using the Kubernetes with Tanka procedure]({{<
...
-    If you have followed the [Tanka Tempo installation example]({{< relref "../setup/tanka" >}}), then the `endpoint` value would be:
+We'll use a [Grafana Agent Helm chart](/docs/agent/latest/flow/setup/install/kubernetes) deployment to send traces to Tempo.
 
-    ```bash
-    distributor.tempo.svc.cluster.local:4317
-    ```
+To do this, you need to create a configuration that can be used by the Agent to receive and export traces in OTLP protobuf format.
 
-1. Apply the ConfigMap with:
+Create a new `values.yaml` file which we'll use as part of the Agent install.
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/set-up-test-app.md:31]:
```suggestion
1. Create a new `values.yaml` file which we'll use as part of the Agent install.
```

File: docs/sources/tempo/setup/set-up-test-app.md [hunk 3]
```diff
@@ -15,110 +15,102 @@ For example, if you [set up Tempo using the Kubernetes with Tanka procedure]({{<
...
 
-    ```bash
-    kubectl apply --namespace default -f agent.yaml
-    ```
+1. Create the file by using `touch`.
+   ```bash
+   touch values.yaml
```
[the remark this answers is not in the document]
Comment (knylander-grafana) [reply docs/sources/tempo/setup/set-up-test-app.md:35]: I added the "Create the file..." sentence as an interpretation of the actual command. Maybe I got it wrong? 

@hedss is it reasonable to say Create a `values.yaml` file or open an existing one used with Tempo. and then delete touch values.yaml?

Comment (knylander-grafana) [inline docs/sources/tempo/setup/set-up-test-app.md:35]:
```suggestion
1. Create a values.yaml file or open an existing one used with Tempo
```

File: docs/sources/tempo/setup/set-up-test-app.md [hunk 4]
```diff
@@ -15,110 +15,102 @@ For example, if you [set up Tempo using the Kubernetes with Tanka procedure]({{<
...
+   ```
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/set-up-test-app.md:36]:
```suggestion
```

File: docs/sources/tempo/setup/set-up-test-app.md [hunk 5]
```diff
@@ -15,110 +15,102 @@ For example, if you [set up Tempo using the Kubernetes with Tanka procedure]({{<
...
 
-1. Navigate to **Configuration ≫ Data Sources**.
+1. Navigate to **Connections** > **Data Sources**.
 
 1. Click on **Add data source**.
 
 1. Select **Tempo**.
 
-1. Set the URL to `http://<TEMPO-HOST>:<HTTP-LISTEN-PORT>/`, filling in the path to your gateway and the configured HTTP API prefix. If you have followed the [Tanka Tempo installation example]({{< relref "../setup/tanka.md" >}}), this will be: `http://query-frontend.tempo.svc.cluster.local:3200/`
+1. Set the URL to `http://<TEMPO-DISTRIBUTOR-SERVICE>:<HTTP-LISTEN-PORT>/`, filling in the path to Tempo's query frontend service, and the configured HTTP API prefix. If you have followed the [Deploy Tempo with Helm installation example]({{< relref "../setup/helm-chart.md" >}}), the query frontend service's URL will look something like this: `http://tempo-cluster-query-frontend.<namespace>.svc.cluster.local:3100`
```
[the remark this answers is not in the document]
Comment (knylander-grafana) [reply docs/sources/tempo/setup/set-up-test-app.md:108]: Good catch. We changed from distributor to query while we were testing. Looks like we missed this. I'll fix it. 
```suggestion
1. Set the URL to `http://<TEMPO-QUERY-FRONTEND-SERVICE>:<HTTP-LISTEN-PORT>/`, filling in the path to Tempo's query frontend service, and the configured HTTP API prefix. If you have followed the [Deploy Tempo with Helm installation example]({{< relref "../setup/helm-chart.md" >}}), the query frontend service's URL will look something like this: `http://tempo-cluster-query-frontend.<namespace>.svc.cluster.local:3100`
```

File: docs/sources/tempo/setup/set-up-test-app.md [hunk 6]
```diff
@@ -129,35 +121,66 @@ For more information, refer to [Tempo in Grafana]({{< relref "../getting-started
 Next, you can use [OpenTelemetry `telemetrygen`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/cmd/telemetrygen) to generate tracing data to test your Tempo installation.
 
 In the following instructions we assume the endpoints for both the Grafana Agent and the Tempo distributor are those described above, for example:
-* `grafana-agent-traces.default.svc.cluster.local` for Grafana Agent
-* `distributor.tempo.svc.cluster.local` for the Tempo distributor
-Replace these appropriately if you have altered the endpoint targets.
+* `grafana-agent.grafana-agent.svc.cluster.local` for Grafana Agent
+* `tempo-cluster-distributor.tempo.svc.cluster.local` for the Tempo distributor
```
[the remark this answers is not in the document]
Comment (knylander-grafana) [reply docs/sources/tempo/setup/set-up-test-app.md:125]:
```suggestion
  * `tempo-cluster-query-frontend.tempo.svc.cluster.local` for the Tempo distributor
```

**Comment by knylander-grafana** [comment]
> Couple o thoughts. I didn't test any of the actual config or steps. Trust these are correct.

Thank you. You caught some of the places where we had changed from distributor to query but didnt' get all of the changes in the text. 

I've asked Heds to do a review to verify that things are correct. When we wrote the doc, he used the configs to make sure that they worked.

File: docs/sources/tempo/setup/set-up-test-app.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hedss) [reply docs/sources/tempo/setup/set-up-test-app.md:35]: Yep, we should do this.

File: docs/sources/tempo/setup/set-up-test-app.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (hedss) [reply docs/sources/tempo/setup/set-up-test-app.md:108]: Yeah, this was snow-blindness, we'd changed it in the example but not the main text. :)

File: docs/sources/tempo/setup/set-up-test-app.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (hedss) [reply docs/sources/tempo/setup/set-up-test-app.md:125]: This isn't for querying though, this is for pushing traces directly at Tempo, so we need the distributor. Am I missing something here?

File: docs/sources/tempo/setup/set-up-test-app.md [hunk 7]
```diff
@@ -15,110 +15,99 @@ For example, if you [set up Tempo using the Kubernetes with Tanka procedure]({{<
...
-1. Apply the ConfigMap with:
+## Configure Grafana Agent Flow to remote-write to Tempo
+
+We'll use a [Grafana Agent Helm chart](/docs/agent/latest/flow/setup/install/kubernetes) deployment to send traces to Tempo.
+
+To do this, you need to create a configuration that can be used by the Agent to receive and export traces in OTLP protobuf format.
+
+1. Create a new `values.yaml` file which we'll use as part of the Agent install.
+
+1. Create a new `values.yaml` file which we'll use as part of the Agent install.
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/set-up-test-app.md:33]:
```suggestion
1. Create a new `values.yaml` file which we'll use as part of the Agent install.

```

File: docs/sources/tempo/setup/set-up-test-app.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (joe-elliott) [reply docs/sources/tempo/setup/set-up-test-app.md:121]: yup. agree. this is the write path. i got confused on this one.

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**